lib/service/invoices.go

```diff
@@ -196,23 +189,46 @@ func (svc *LndhubService) PayInvoice(ctx context.Context, invoice *models.Invoic
 
 	// The payment was successful.
 	invoice.Preimage = paymentResponse.PaymentPreimageStr
-	invoice.State = common.InvoiceStateSettled
-	invoice.SettledAt = schema.NullTime{Time: time.Now()}
+	err = svc.HandleSuccessfulPayment(context.Background(), invoice)
+	return &paymentResponse, err
+}
 
-	_, err = tx.NewUpdate().Model(invoice).WherePK().Exec(ctx)
+func (svc *LndhubService) HandleFailedPayment(ctx context.Context, invoice *models.Invoice, entryToRevert models.TransactionEntry, failedPaymentError error) error {
+	// add transaction entry with reverted credit/debit account id
+	entry := models.TransactionEntry{
+		UserID:          invoice.UserID,
+		InvoiceID:       invoice.ID,
+		CreditAccountID: entryToRevert.DebitAccountID,
+		DebitAccountID:  entryToRevert.CreditAccountID,
+		Amount:          invoice.Amount,
+	}
+	_, err := svc.DB.NewInsert().Model(&entry).Exec(ctx)
 	if err != nil {
-		tx.Rollback()
-		return nil, err
+		svc.Logger.Errorf("Could not insert transaction entry user_id:%v invoice_id:%v", invoice.UserID, invoice.ID)
+		return err
```

Comment:
those errors are still sent to Sentry? or should we do that explicitly here? 
We say above we run those in a Background context to ignore the request context. 

I just think we have to make sure that we have notifications for such exceptions. This one would require some admin to do something.